File: ProgrammingFundamentalsExercises/ArraysAndListsExercises/01.LargestCommonEnd/LargestCommonEnd.cs
using System;
using System.Collections.Generic;

class LargestCommonEnd
{
    static void Main(string[] args)
    {
        string[] inputOne = Console.ReadLine().Split(' ');
        string[] inputTwo = Console.ReadLine().Split(' ');
        string[] longerArray = new string[Math.Max(inputOne.Length, inputTwo.Length)];
        string[] shorterArray = new string[Math.Min(inputOne.Length, inputTwo.Length)];

        if (inputOne.Length >= inputTwo.Length)
        {
            longerArray = inputOne;
            shorterArray = inputTwo;
        }
        else 
        {
            longerArray = inputTwo;
            shorterArray = inputOne;
        }

        int countLeft = 0;

        for (int i = 0; i < shorterArray.Length; i++)
        {
            if (shorterArray[i] == longerArray[i])
            {
                countLeft++;
            }
            else
            {
                break;
            }
        }

        int longerArrayPosition = longerArray.Length - 1;
        int countRight = 0;

        for (int i = shorterArray.Length - 1; i >= 0; i--)
        {
            if (shorterArray[i] == longerArray[longerArrayPosition])
            {
                countRight++;
                longerArrayPosition--;
            }
        }

        Console.WriteLine(Math.Max(countRight, countLeft));
    }
}

File: JavaTesting/src/LargestCommonEnd.java
import java.util.Scanner;

public class LargestCommonEnd {
    public static void main(String[] args) {
        Scanner console = new Scanner(System.in);

        String[] inputOne = console.nextLine().split("\\s");
        String[] inputTwo = console.nextLine().split("\\s");

        int frontCount = getEndCount(inputOne, inputTwo);

        String[] inputOneReversed = new String[inputOne.length];
        String[] inputTwoReversed = new String[inputTwo.length];

        int j = 0;
        for (int i = inputOne.length - 1; i >= 0; i--) {
            inputOneReversed[j] = inputOne[i];
            j++;
        }

        j = 0;
        for (int i = inputTwo.length - 1; i >= 0; i--) {
            inputTwoReversed[j] = inputTwo[i];
            j++;
        }

        int backCount = getEndCount(inputOneReversed, inputTwoReversed);

        System.out.println(Math.max(frontCount, backCount));
    }

    public static int getEndCount(String[] arrayOne, String[] arrayTwo) {
        int count = 0;

        for (int i = 0; i < Math.min(arrayOne.length, arrayTwo.length); i++) {
            if (arrayOne[i].equals(arrayTwo[i])) {
                count++;
            } else {
                break;
            }
        }

        return count;
    }
}